src/components/Palmid/Palmid.tsx

```diff
@@ -0,0 +1,187 @@
+            if (!isReportReady && !isCheckReportTimedOut) {
+                // first call
+                checkReport()
+
+                //later calls
+                interval = setInterval(() => {
+                    checkReport()
+                }, REQUEST_INTERVAL)
+                return () => clearInterval(interval)
+            } else {
+                // stop the api call to check for report
+                clearInterval(interval)
+            }
+        }
+    }, [fastaHash, isReportReady, isCheckReportTimedOut])
+
+    // Request timeout after 5 minutes
+    useEffect(() => {
+        if (isCheckReportTimedOut && !isReportReady) {
+            setShowIframe(false)
+        }
+    }, [isCheckReportTimedOut, isReportReady])
+
+    useEffect(() => {
+        let hashExists = false
+        hashExists = loadUrlHash()
+        if (hashExists) {
+            setShowIframe(true)
+            setIsFastaCollapsed(!isFastaCollapsed)
+        }
+    }, [])
+
+    return (
+        <>
+            <h1 className='text-3xl m-2 font-bold text-center '>palmID: Viral-RdRP Analysis</h1>
+
+            <button
+                className='m-auto rounded bg-blue-500 hover:bg-blue-700 text-white font-bold py-0.5 collapse-button'
+                onClick={() => setIsFastaCollapsed(!isFastaCollapsed)}>
+                Sequence Submission
+            </button>
+            <div
+                id='fastaSubmission'
+                className={`m-4 p-4 collapse-content ${
+                    !isFastaCollapsed ? 'collapsed' : 'expanded'
+                }`}
+                aria-expanded={isFastaCollapsed}>
+                <p className='my-3'>Sequence, in FASTA format</p>
+                <textarea
+                    className='border-2 focus:ring-1 rounded focus: outline-none resize-none  mb-2 p-2'
+                    id='fastaInput'
+                    value={fastaInput}
+                    rows={4}
+                    cols={72}
+                    placeholder='>Enter your sequence (DNA / Protein)'
+                    aria-required='true'
+                    onChange={(e) => {
+                        setFastaInput(e.target.value)
+                    }}></textarea>
+                <div className='white-space: pre-line'>
+                    <p>{`Parsed FASTA: `}</p>
+                    <pre>
+                        {parsedFastaSequenceHeader}
+                        <br></br>
+                        {parsedFastaSequenceText}
+                    </pre>
+                    <code> hash: {fastaHash}</code>
+                </div>
+                <br></br>
+                <div>
+                    <button
+                        className='w-300 m-auto rounded bg-blue-500 hover:bg-blue-700 text-white font-bold py-1 px-4 disabled:cursor-not-allowed disabled:bg-gray-500 disabled:text-gray-200'
+                        disabled={!fastaInput}
+                        onClick={async () => {
```

Comment:
I ran a quick test on https://palmid.serratus.io/palmid and got the expected timeout. However, when clicking to analyze a new sequence, the state of `isCheckReportTimedOut` isn't reset so it doesn't show `load_palmid.gif` and appears "stuck" on the timeout result. There should be a way to reset that without `clear()` since that applies to everything.

Reply:
Good edge-case catch! I added a timer reset when the Analyze button is pressed. Alternative thought is to gray-out the analyze button once it's pressed so reports are not made "one on top of another". Each refresh / clear == 1 report.